website/blog/2024/04-23-scrapy-vs-crawlee/index.md

```diff
@@ -0,0 +1,368 @@
+
+<!--truncate-->
+
+## Feature comparison
+
+There are a lot of things that we can compare between Scrapy and Crawlee. This article will be the first part of a series comparing Scrapy and Crawlee on various parameters. 
+
+We will compare both libraries on various parameters, starting with language and development environments and essential features that make the scraping process easy for developers, like autoscaling, headless browsing, queue management, etc. 
+
+
+
+## Language and development environments:
+
+Scrapy is written in Python, making it easier for the data science community to integrate it with various tools with Python. While Scrapy offers very detailed documentation, for first-timers, sometimes it takes work to start with Scrapy. One of the reasons why it is considered not so beginner-friendly is its [complex architecture](https://docs.scrapy.org/en/latest/topics/architecture.html), which consists of various components like spiders, middleware, item pipelines, and settings. For beginners, learning all of these can be a time-consuming task.
+
+On the other hand, Crawlee is one of the few web scraping and automation libraries that supports JavaScript and TypeScript. Crawlee supports CLI as Scrapy does, but the difference that it makes very easy for beginners to start with is their [pre-built templates](https://github.com/apify/crawlee/tree/master/packages/templates/templates) in TypeScript and JavaScript supporting Playwright and Puppeteer. It helps beginners to get a quick understanding of the file structure and how it works.
+
+## Working with JavaScript-heavy websites
+
+One problem that occurs with Scrapy is that it needs to work better with JavaScript-heavy websites. For example, we will try to scrape a JavaScript-heavy website; we will try scraping `showName,` `seasons,` and `synopsis` from one of the famous Netflix famous shows pages, [Stranger Things](https://www.netflix.com/in/title/80057281).
+
+With Scrapy, when we try to scrape all of the above things by installing `scrapy` and then creating a spider with the following script:
+
+```python
+import scrapy
+
+class NetflixSpider(scrapy.Spider):
+    name = 'netflix'
+    allowed_domains = ['netflix.com']
+    start_urls = ['https://www.netflix.com/title/80057281']
+
+    def parse(self, response):
+        title = response.css('.title-title::text').get()
+        seasons = response.css('.test_dur_str::text').get()
+        about = response.css('div.title-info-synopsis::text').get()
+
+        yield {
+            'title': title,
+            'seasons': seasons,
+            'about': about
+        }
+```
+
+The GET produces a `200` result; still, we get an empty array in the result tab because Scrapy could not render the JavaScript.
+
+When we scrape the same parameters from the same website using Crawlee using the following script:
+
+```js
+const { PlaywrightCrawler, Dataset } = require('crawlee');
+
+async function pageFunction({ page, request }) {
```

Comment:
also the way you have it now wont be typed properly, in fact this would fail to compile under strict TS config. you should inline this completely instead of defining a separate function, that way the type is inferred. or use [the router](https://crawlee.dev/api/playwright-crawler/function/createPlaywrightRouter).